client/x/evmstaking/keeper/withdrawal_queue_test.go

```diff
@@ -0,0 +1,246 @@
+package keeper_test
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/piplabs/story/client/x/evmstaking/types"
+)
+
+var (
+	delAddr     = "story1hmjw3pvkjtndpg8wqppwdn8udd835qpan4hm0y"
+	valAddr     = "storyvaloper1hmjw3pvkjtndpg8wqppwdn8udd835qpaa6r6y0"
+	evmAddr     = common.HexToAddress("0x131D25EDE18178BAc9275b312001a63C081722d2")
+	withdrawals = []types.Withdrawal{
+		types.NewWithdrawal(1, delAddr, valAddr, evmAddr.String(), 100),
+		types.NewWithdrawal(2, delAddr, valAddr, evmAddr.String(), 200),
+		types.NewWithdrawal(3, delAddr, valAddr, evmAddr.String(), 300),
+	}
+)
+
+func (s *TestSuite) initQueue() {
+	err := s.EVMStakingKeeper.WithdrawalQueue.Initialize(s.Ctx)
+	s.NoError(err)
+	s.Equal(uint64(0), s.EVMStakingKeeper.WithdrawalQueue.Len(s.Ctx))
+}
+
+func (s *TestSuite) addWithdrawals(withdrawals []types.Withdrawal) {
+	for _, w := range withdrawals {
+		err := s.EVMStakingKeeper.AddWithdrawalToQueue(s.Ctx, w)
+		s.NoError(err)
+	}
+}
+
+func (s *TestSuite) TestAddWithdrawalToQueue() {
+	s.initQueue()
+
+	// Add a withdrawal to the queue
+	withdrawal := types.NewWithdrawal(1, delAddr, valAddr, evmAddr.String(), 100)
+	err := s.EVMStakingKeeper.AddWithdrawalToQueue(s.Ctx, withdrawal)
+	s.NoError(err)
+
+	// Check the withdrawal is in the queue
+	s.Equal(uint64(1), s.EVMStakingKeeper.WithdrawalQueue.Len(s.Ctx))
+	elem, err := s.EVMStakingKeeper.WithdrawalQueue.Get(s.Ctx, 0)
+	s.NoError(err)
+	s.Equal(withdrawal, elem)
+}
+
+func (s *TestSuite) TestDequeueEligibleWithdrawals() {
+	tcs := []struct {
+		name        string
+		maxDequeue  uint32
+		expectedLen int
+		expected    []types.Withdrawal
+	}{
+		{
+			name:        "Dequeue 1 withdrawal",
+			maxDequeue:  1,
+			expectedLen: 1,
+			expected:    withdrawals[:1],
+		},
+		{
+			name:        "Dequeue 2 withdrawals",
+			maxDequeue:  2,
+			expectedLen: 2,
+			expected:    withdrawals[:2],
+		},
+		{
+			name:        "Dequeue more than available",
+			maxDequeue:  10,
+			expectedLen: len(withdrawals),
+			expected:    withdrawals,
+		},
+		{
+			name:        "Dequeue with empty queue",
+			maxDequeue:  3,
+			expectedLen: 0,
+			expected:    []types.Withdrawal{},
+		},
```

Comment:
Add dequeue more than MaxWithdrawalPerBlock

Reply:
@limengformal 
Just to clarify, the `DequeueEligibleWithdrawals` is already designed to respect the `MaxWithdrawalPerBlock` limit. The test cases ensure that even if the queue has more withdrawals than the max limit, only up to `MaxWithdrawalPerBlock` are dequeued.

Could you please provide more details or an example of what you mean by ‘dequeue more than MaxWithdrawalPerBlock’? I want to ensure I’m addressing your concern accurately.

Reply:
Sounds good. I missed that. Looks good now